internal/collector/processor/nodebatcher/node_batcher.go

```diff
@@ -332,37 +256,39 @@ func (bt *bucketTicker) start() {
 						// re-delete just in case
 						delete(bt.nodes, nbKey)
 						continue
-					}
-					// atomic.LoadPointer only takes unsafe.Pointer interfaces. We do not use unsafe
-					// to skirt around the golang type system.
-					ptr := (*unsafe.Pointer)(unsafe.Pointer(&nb.currBatch))
-					b := (*batch)(atomic.LoadPointer(ptr))
-					if atomic.LoadInt64(&nb.lastSent)+nb.timeout.Nanoseconds() > time.Now().UnixNano() {
-						// In this case, we sent more recently than the timeout, so do nothing
-						continue
-					} else if b.getCurrentItemCount() > 0 {
-						// If the batch is non-empty, go ahead and cut it
-						statsTags := processor.StatsTagsForBatch(
-							nb.parent.name, processor.ServiceNameForNode(nb.node), nb.spanFormat,
-						)
-						stats.RecordWithTags(context.Background(), statsTags, statTimeoutTriggerSend.M(1))
-						nb.cutBatch(b)
 					} else {
-						cyclesUntouched := atomic.AddUint32(&nb.cyclesUntouched, 1)
-						if cyclesUntouched > nb.parent.removeAfterCycles {
-							nb.parent.removeBucket(nbKey)
-							delete(bt.nodes, nbKey)
-							// At this point, add() should no longer be callable from the parent,
-							// so mark this nodeBatcher as dead (for adds that have already been called and are racing)
-							// and try to cut a final batch (just in case something was added in between
-							// the above if and marking this nodeBatcher as dead)
-							atomic.StoreUint32(&nb.dead, nodeStatusDead)
-							// Reload batch in case `add` loaded a new batch that was then added to.
-							currBatchPtr := (*unsafe.Pointer)(unsafe.Pointer(&nb.currBatch))
-							b = (*batch)(atomic.LoadPointer(currBatchPtr))
-							if b.getCurrentItemCount() > 0 {
-								nb.cutBatch(b)
+						nb.mu.Lock()
+						if len(nb.items) > 0 {
```

Comment:
Can we have some different functions for this part since the indentation is going crazy here and is very hard to read?

Reply:
done